app/assets/javascripts/lib/models/chart_list.coffee

```diff
@@ -175,7 +181,23 @@ class @ChartList extends Backbone.Collection
           ignore_lock: true
         })
 
-    $.when.apply(null, @chart_requests).done((d) =>
+    # The @chart_requests are jqXHR objects returned by $.ajax().
+    # Here, we want to wait until all Ajax requests have returned, no matter
+    # their success. But Promise.all (and similar constructs like $.when)
+    # has a fast-failure feature if one of the requests fails.
+    # Hence, we have to wrap the jqXHRs in new Promises which will
+    # always return successfully - after all, we are not interested in the
+    # success of the requests anyways (since success is handled directly in
+    # callbacks to the Ajax calls).
+    Promise.all(@chart_requests.map((request) ->
```

Comment:
I had to change the existing code anyways because the fast-failing feature of `$.when` prevented the successful calls from triggering the `done` function - or if we used `$.always` instead of `$.when`, the successful calls would not yet be back and therefore the charts would not be created yet (I tried this locally).
And then the JavaScript-native Promise-machinery seemed like the most natural choice to do this. But I am confident that I can change this to JQuery-Deferred-machinery.